### PR TITLE
Update logging in our deployment controllers

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/13761

While here this also fixes https://github.com/openshift/origin/issues/11854

@ncdc @mfojtik

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13067 + https://github.com/openshift/origin/issues/13770 + https://github.com/openshift/origin/issues/9999

**Comment by kargakis** [comment]
@soltysh can you have a look since @mfojtik is out today?

**Comment by soltysh** [comment]
LGTM [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13799 + https://github.com/openshift/origin/issues/13067 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13802 + https://github.com/openshift/origin/issues/8427 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13811 [merge]

File: pkg/deploy/controller/generictrigger/factory.go [hunk 1]
```diff
@@ -229,12 +230,11 @@ func (c *DeploymentTriggerController) work() bool {
...
 	obj, exists, err := c.dcLister.Indexer.GetByKey(key)
 	if err != nil {
-		glog.Infof("Unable to retrieve deployment config %q from store: %v", key, err)
-		c.queue.Add(key)
+		c.queue.AddRateLimited(key)
 		return nil, err
 	}
 	if !exists {
-		glog.Infof("Deployment config %q has been deleted", key)
+		glog.V(4).Infof("Deployment config %q has been deleted", key)
```
Comment (mfojtik) [inline pkg/deploy/controller/generictrigger/factory.go]: nit: DeploymentConfig (to be consistent)

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13812 [merge]

**Comment by kargakis** [comment]
This is ridiculous

https://github.com/openshift/origin/issues/13814 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/12072 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13823 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13823 [merge]

**Comment by kargakis** [comment]
#13823 [merge]

**Comment by smarterclayton** [comment]
[merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13814 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/13846 [merge]

**Comment by kargakis** [comment]
#13846 [merge]

**Comment by kargakis** [comment]
[merge]

**Comment by kargakis** [comment]
I don't know but this looks badly broken

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/9397 [merge]

**Comment by mfojtik** [comment]
very nice.

**Comment by smarterclayton** [comment]
Wow

**Pull request merged**